tsl/src/compression/compress_utils.c

```diff
@@ -138,14 +139,11 @@ compresschunkcxt_init(CompressChunkCxt *cxt, Cache *hcache, Oid hypertable_relid
 						 " the timescaledb.compress option.")));
 	}
 	compress_ht = ts_hypertable_get_by_id(srcht->fd.compressed_hypertable_id);
-	if (compress_ht == NULL)
-		ereport(ERROR, (errcode(ERRCODE_INTERNAL_ERROR), errmsg("missing compress hypertable")));
+	ASSERT_OR_ERROR(compress_ht != NULL, "missing compress hypertable");
 	/* user has to be the owner of the compression table too */
 	ts_hypertable_permissions_check(compress_ht->main_table_relid, GetUserId());
 
-	if (!srcht->space) /* something is wrong */
-		ereport(ERROR,
-				(errcode(ERRCODE_INTERNAL_ERROR), errmsg("missing hyperspace for hypertable")));
+	ASSERT_OR_ERROR(srcht->space, "missing hyperspace for hypertable");
```

Comment:
This is an error too.  Some other session probably messed things up. A core dump or back trace  does not help.